### ISSUE TITLE
[Bug] Handle leading 0 in semver

### ISSUE THREAD

**Issue by prankstr** [democratic-csi/democratic-csi#365]
I ran into issues very similar to #326 when upgrading to TrueNAS SCALE 24.04-BETA.1. Since the log indicated that deprecated API parameters was used I suspected that the version check failed for some reason.

In my case it turns out semver doesn't like the 0 in .04 and `semver.coerce` returns null for 24.04. Issue about leading 0s on semver here: https://github.com/npm/node-semver/issues/232

I fixed it by adding the suggested "loose" to semver.coerce:
```
const truenasVersion = semver.coerce(
  await httpApiClient.getSystemVersionMajorMinor(), { loose: true }
);
```

This translates 24.04 to 24.4.0 which made it use the correct parameters in the API call. If you think it's a feasible way forward I can make a PR. Might also be good to check if it's a valid semver after the coerce and log something to the user if it's not.

**Comment by travisghansen** [comment]
Yeah go ahead and make a PR targeted at the `next` branch. Thanks!

**Comment by travisghansen** [comment]
https://github.com/democratic-csi/democratic-csi/commit/15b120c3cc75d64fa164a9bea7b5c107f95eee21

**Comment by travisghansen** [comment]
Fix should be released in `v1.9.0`. Reopen if you see the behavior continue after upgrading.
